src/parser/transform/statement/transform_create_enum.cpp

```diff
@@ -17,14 +18,35 @@ vector<string> ReadPgListToString(duckdb_libpgquery::PGList *column_list) {
 	return result;
 }
 
+Vector ReadPgListToVector(duckdb_libpgquery::PGList *column_list, idx_t &size) {
+	Vector result(LogicalType::VARCHAR);
+	if (!column_list) {
+		return result;
+	}
+	// First we discover the size of this list
+	for (auto c = column_list->head; c != nullptr; c = lnext(c)) {
+		size++;
+	}
+	if (size > STANDARD_VECTOR_SIZE) {
```

Comment:
We can allocate with the correct size immediately here (`Vector result(LogicalType::VARCHAR, size)`).